tds.go

```diff
@@ -691,9 +693,240 @@ func splitConnectionString(dsn string) (res map[string]string) {
 	return res
 }
 
+// Splits a URL in the ODBC format
+func splitConnectionStringOdbc(dsn string) (map[string]string, error) {
+	res := map[string]string{}
+
+	type parserState int
+	const (
+		// Between pairs
+		parserStateBetweenPairs parserState = iota
+
+		// Inside a key
+		parserStateKey parserState = iota
+
+		// Beginning of a value. May be bare or braced
+		parserStateBeginValue parserState = iota
+
+		// Inside a bare value
+		parserStateBareValue parserState = iota
+
+		// Inside a braced value
+		parserStateBracedValue parserState = iota
+
+		// A closing brace inside a braced value.
+		// May be the end of the value or an escaped closing brace, depending on the next character
+		parserStateBracedValueClosingBrace parserState = iota
+
+		// After a value. Next character should be a semi-colon or whitespace.
+		parserStateEndValue parserState = iota
+	)
+
+	var state = parserStateBetweenPairs
+
+	var key string
+	var value string
+
+	for i, c := range dsn {
+		switch state {
+		case parserStateBetweenPairs:
+			if c == '=' {
```

Comment:
I'm okay with these if/else statements. I would prefer them to be switch statements:
```
switch {
case c == '=':
    return res, err
case !unicode.IsSpace(c) && c != ';':
    ...
}
```